serde/src/java/org/apache/hadoop/hive/serde2/lazybinary/LazyBinaryTimestamp.java

```diff
@@ -52,6 +55,28 @@ public class LazyBinaryTimestamp extends
    */
   @Override
   public void init(ByteArrayRef bytes, int start, int length) {
-    data.set(bytes.getData(), start);
+    if (!legacyConversionEnabled) {
+      // Hive 3.x default: interpret as UTC
+      data.set(bytes.getData(), start);
+    } else {
+      // Legacy Hive 2.x behavior: interpret as local time
+      data.set(bytes.getData(), start);
+
+      // Convert to java.sql.Timestamp
+      Timestamp ts = data.getTimestamp();
```

Comment:
if that is already handled by other file formats, can't we reuse/extract the conversion logic into util class?